### PR TITLE
[dev-infra][build] Create script to build multiple AWS Codebuild projects

### PR DESCRIPTION
**Pull request by ankushagarwal**
## Summary

* Create a single script `docker/build-aws.sh` which can handle multiple parallel builds and print status messages
* Update cluster-test build script `docker/cluster-test/build.sh` so that it doesnt need a continously running build container for builds
* Update cti to refer to `docker/build-aws.sh` for building
* Fix ct script so that it can parse multiple arguments

## Test Plan

Ran `./docker/build-aws.sh --build-all --version pull/2263` and `./scripts/cti --pr 2263  --run bench` successfully

### CODE REVIEW

File: terraform/templates/cluster-test/ct [hunk 1]
```diff
@@ -44,11 +47,15 @@ done
...
 then
       echo "Using master cluster test image"
-      $(aws ecr get-login --no-include-email --region us-west-2)
       DOCKER_IMAGE="853397791086.dkr.ecr.us-west-2.amazonaws.com/libra_cluster_test:master"
-      docker pull $DOCKER_IMAGE
+else
+      echo "Using $DOCKER_IMAGE image"
 fi
 
+echo "Logging in to ECR repo"
```
Comment (andll) [inline terraform/templates/cluster-test/ct]: Why did you make this change?

File: terraform/templates/cluster-test/ct [hunk 2]
```diff
@@ -12,24 +12,27 @@ while (( "$#" )); do
     -c|--container)
       CONTAINER=$2
       shift 2
+      continue
```
Comment (andll) [inline terraform/templates/cluster-test/ct]: Why is this needed?

File: docker/cluster-test/buildspec.yaml [hunk 3]
```diff
@@ -13,7 +13,7 @@ phases:
     commands:
       - echo Build started on `date`
       - echo Building the Docker image...
-      - docker/cluster-test/build.sh
+      - ephemeral_container=1 docker/cluster-test/build.sh
```
Comment (andll) [inline docker/cluster-test/buildspec.yaml]: Can you give some more details what is different about code build env?

File: terraform/templates/cluster-test/ct (hunk 2 above)
Comment (ankushagarwal) [reply terraform/templates/cluster-test/ct]: When I passed multiple ct script specific flags to ct (eg `ct -c cluster-test-ci --image image-id`), it was only processing the first flag `-c cluster-test-ci` and ignoring the `--image image-id`. With this, the flags are being parsed correctly.

File: terraform/templates/cluster-test/ct (hunk 1 above)
Comment (ankushagarwal) [reply terraform/templates/cluster-test/ct]: We were only doing a `docker pull` for `:master` tag. 

With `cti`, when we start using the `:dev_$USER_pr_1234` tag, we will have to pull that as well because for different builds of the same pr, `cti` will use the same tag and hence I updated this code to always pull.

File: terraform/templates/cluster-test/ct (hunk 2 above)
Comment (andll) [reply terraform/templates/cluster-test/ct]: This is very strange, can you give me a repro?
```
[ec2-user@cluster-test-runner ~]$ ct -c xxx --image image-id
Using image-id image
[ec2-user@cluster-test-runner ~]$ ct --image image-id -c xxx
Using image-id image
```

File: docker/cluster-test/buildspec.yaml (hunk 3 above)
Comment (ankushagarwal) [reply docker/cluster-test/buildspec.yaml]: This command fails to run in the codebuild env `docker run --name libra_cluster_test_builder_container -d -i -t -v $DIR/../..:/libra libra_cluster_test_builder bash`

It gives the error : `the input device is not a TTY`

File: terraform/templates/cluster-test/ct (hunk 2 above)
Comment (ankushagarwal) [reply terraform/templates/cluster-test/ct]: I am unable to repro it, I must've made some mistake earlier. I have updated it to remove the continue statements

File: terraform/templates/cluster-test/ct (hunk 1 above)
Comment (andll) [reply terraform/templates/cluster-test/ct]: I see.
One of usages is `--latest` flag that uses locally compiled latest image, so docker pull will fail on it
I suggest to just add docker pull to `-i` flag

File: docker/cluster-test/buildspec.yaml (hunk 3 above)
Comment (andll) [reply docker/cluster-test/buildspec.yaml]: I see.
It's perfectly fine to just remove `-t` flag for both cases then - this way you won't need ephemeral_container var and I don't think it will hurt anything

File: terraform/templates/cluster-test/ct [hunk 4]
```diff
@@ -5,7 +5,9 @@ set -e
 
 DOCKER_IMAGE=""
 WAIT_TO="45m"
-export CONTAINER="cluster-test-$RANDOM"
+CONTAINER="cluster-test-$RANDOM"
```
Comment (andll) [inline terraform/templates/cluster-test/ct]: I think there was some reason to export it here(afaik it was required for one of lines below), lets keep it?

Comment (ankushagarwal) [reply terraform/templates/cluster-test/ct]: Fixed

File: scripts/cti [hunk 5]
```diff
@@ -60,4 +60,4 @@ echo " * ssh-peer"
 echo " * tail -f /data/libra/libra.log"
 echo "**********"
 
-ssh -t ssh.${WORKSPACE}.aws.hlw3truzy4ls.com ssh -t -i /libra_rsa ec2-user@ct.priv.${WORKSPACE}.aws.hlw3truzy4ls.com ct -c cluster-test-ci --deploy $TAG $*
+ssh -t ssh.cluster-test-ci.aws.hlw3truzy4ls.com ssh -i /libra_rsa ec2-user@ct.priv.cluster-test-ci.aws.hlw3truzy4ls.com ct -c cluster-test-ci --image "853397791086.dkr.ecr.us-west-2.amazonaws.com/libra_cluster_test:$TAG" --deploy $TAG $*
```
Comment (andll) [inline scripts/cti]: FYI - looks like this change also partially reverted change that introduced -W flag (`${WORKSPACE}` instead of `cluster-test-ci `). I'll fix in separate PR but lets be more careful when merging next time

Comment (ankushagarwal) [reply scripts/cti]: Ahh, sorry about that.
